### DISCUSSION TITLE
Error running Kaggle example: Forward() takes 2 positional arguments but 3 were given

### DISCUSSION THREAD

**Discussion by TimbusCalin** [rwightman/efficientdet-pytorch#153] [Q&A]
I am using PyTorch 1.7.1 and have some problems with the fit logic.

More precisely, those lines of code below.

What exactly should I do in order to successfully train?

I followed the tutorial here and the bug was solved with the "target_res" dictionary (https://www.kaggle.com/shonenkov/training-efficientdet) but in my case the initial error was` Forward() takes 2 positional arguments but 4 were given`. Of course, replacing two lists with a dict() like below led me to "3" instead of "4"

Would anyone be so kind as to point me in the right direction?

Thank you

```
            images = torch.stack(images)
            images = images.to(self.device).float()
            batch_size = images.shape[0]

            boxes = [target['boxes'].to(self.device).float() for target in targets]
            labels = [target['labels'].to(self.device).float() for target in targets]

            target_res = {}
            target_res['bbox'] = boxes
            target_res['cls'] = labels

            self.optimizer.zero_grad()
           
            #It fails here
            outputs = self.model(images, target_res)
            loss = outputs['loss']
            loss.backward()

            summary_loss.update(loss.detach().item(), batch_size)
            self.optimizer.step()
```

**Comment by TimbusCalin** [comment]
My fault --- I did not instantiate the model correctly.

For those interested, have a look here:


def get_net():
    config = get_efficientdet_config('tf_efficientdet_d0')

    config.image_size = [512, 512]
    config.norm_kwargs = dict(eps=.001, momentum=.01)

    net = EfficientDet(config, pretrained_backbone=False)

    net.reset_head(num_classes=1)
    net.class_net = HeadNet(config, num_outputs=config.num_classes)
    return DetBenchTrain(net, config)

**Comment by rwightman** [comment]
@TimbusCalin yeah, not a bug but just examples that are out of date with the latest here... I'll move this to discussion as Q related to this have come up before
